lib/plotting_functions.py

```diff
@@ -121,49 +149,89 @@ def plot_map_and_save(wks, mdlfld, obsfld, diffld, **kwargs):
     else:
+        levels1 = np.linspace(minval, maxval, 12)   
         norm1 = mpl.colors.Normalize(vmin=minval, vmax=maxval)
-        cmap1 = 'coolwarm'
 
-    if 'cnLevels' in kwargs:
-        levels1 = kwargs.pop(cnLevels)
+
+    if ('colormap' not in kwargs) and ('contour_levels' not in kwargs):
+        if ((minval < 0) and (0 < maxval)) and mplv > 2:
+            norm1 = normfunc(vmin=minval, vmax=maxval, vcenter=0.0)
+        else:
+            norm1 = mpl.colors.Normalize(vmin=minval, vmax=maxval)
+
+    # Difference options -- Check in kwargs for colormap and levels
+    if "diff_colormap" in kwargs:
+        cmapdiff = kwargs["diff_colormap"]
+    else:
+        cmapdiff = 'coolwarm'
+    
+    if "diff_contour_levels" in kwargs:
+        levelsdiff = kwargs["diff_contour_levels"]  # a list of explicit contour levels
+    elif "diff_contour_range" in kwargs:
+            assert len(kwargs['diff_contour_range']) == 3  # start, stop, step
+            levelsdiff = np.arange(*kwargs['diff_contour_range'])
     else:
-        levels1 = np.linspace(minval, maxval, 12)
-    # set a symmetric color bar for diff:
-    absmaxdif = np.max(np.abs(diffld))
-    # set levels for difference plot:
-    levelsd = np.linspace(-1*absmaxdif, absmaxdif, 12)
+        # set a symmetric color bar for diff:
+        absmaxdif = np.max(np.abs(diffld))
+        # set levels for difference plot:
+        levelsdiff = np.linspace(-1*absmaxdif, absmaxdif, 12)
+
+    # color normalization for difference
+    if ((np.min(levelsdiff) < 0) and (0 < np.max(levelsdiff))) and mplv > 2:
+        normdiff = normfunc(vmin=np.min(levelsdiff), vmax=np.max(levelsdiff), vcenter=0.0)
+    else:
+        normdiff = mpl.colors.Normalize(vmin=np.min(levelsdiff), vmax=np.max(levelsdiff))
+    
+
+    # extract any MPL kwargs that should be passed on:
+    if 'mpl' in kwargs:
+        subplots_opt = kwargs['mpl'].get('subplots')
+        contourf_opt = kwargs['mpl'].get('contourf')
+        colorbar_opt = kwargs['mpl'].get('colorbar')
+    else:
+        subplots_opt = contourf_opt = colorbar_opt = {}
 
-    fig, ax = plt.subplots(figsize=(6,12), nrows=3, subplot_kw={"projection":ccrs.PlateCarree()})
+    fig, ax = plt.subplots(figsize=(6,12), nrows=3, subplot_kw={"projection":ccrs.PlateCarree()}, **subplots_opt)
     img = [] # contour plots
     cs = []  # contour lines
     cb = []  # color bars
 
     for i, a in enumerate(wrap_fields):
+
         if i == len(wrap_fields)-1:
-            levels = levelsd #Using 'levels=12' casued len() error in mpl. -JN
-            #Only use "vcenter" if "matplotlib" version is greater than 2:
-            if(mplv > 2):
-                norm = normfunc(vmin=-1*absmaxdif, vcenter=0., vmax=absmaxdif)
-            else:
-                norm = normfunc(vmin=-1*absmaxdif, vmax=absmaxdif)
-            cmap = 'coolwarm'
+            levels = levelsdiff #Using 'levels=12' casued len() error in mpl. -JN
```

Comment:
We can probably remove my comment here, as it clearly seems to be working now.

Reply:
I'm removing it, but we should check on setting an integer number of levels that are automatically selected by mpl.